Makefile

```diff
@@ -64,10 +70,30 @@ clusterctl:
 		-o bin/clusterctl \
 		cmd/clusterctl/main.go
 
-test: unit functional
+test: unit functional generate_yaml_test
 
 check: depend fmt vet lint
 
+generate_yaml_test:
+ifndef HAS_YQ
+		go get github.com/mikefarah/yq
+		echo "installing yq"
+endif
+
+ifndef HAS_KUSTOMIZE
+		# for now, higher version has some problem so we stick to 1.0.11
+		wget https://github.com/kubernetes-sigs/kustomize/releases/download/v1.0.11/kustomize_1.0.11_linux_amd64
+		mv kustomize_1.0.11_linux_amd64 /usr/local/bin/kustomize
+		chmod +x /usr/local/bin/kustomize
+endif
+		
+	# Create a dummy file for test only
+	echo 'clouds' > dummy-clouds-test.yaml
+	$(GENERATE_YAML_PATH)/$(GENERATE_YAML_EXEC) -f dummy-clouds-test.yaml openstack ubuntu $(GENERATE_YAML_TEST_FOLDER)
```

Comment:
How to check if the command is succeed or not?

Reply:
nothing to check ,just run the command to make sure we can generate yaml
for example , #321 actually can be found if we have this test